### PR TITLE
Add `_generate_next_value_` to `StrEnum`

### PR DESCRIPTION
**Pull request by sobolevn**
This change is enough to (at least partially) fix https://github.com/python/typeshed/issues/10384

Why? Because `mypy` already had this plugin in place: https://github.com/python/mypy/blob/b995e1620789a3ab2fc5dbcf0698a9077b0f5731/mypy/plugins/enums.py#L85-L90

Source: https://github.com/python/cpython/blob/319de0b578ec3dec8d30610caee3aa93f51060c7/Lib/enum.py#L1279

So, adding it locally produces:

```python
from enum import StrEnum, auto

class A(StrEnum):
    a = auto()

reveal_type(type(A.a))  # N: Revealed type is "Type[Literal[ex.A.a]?]"
reveal_type(type(A.a.value))  # N: Revealed type is "Type[builtins.str]"
```

Closes https://github.com/python/typeshed/issues/10384

### CODE REVIEW

File: tests/stubtest_allowlists/py311.txt [hunk 1]
```diff
@@ -17,6 +17,7 @@ configparser.LegacyInterpolation.__init__
 configparser.ParsingError.filename
 enum.Enum.__init__
 enum.Enum._generate_next_value_
+enum.StrEnum._generate_next_value_
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py311.txt:20]: Can you add a comment here to say why we're allowlisting it?

**Review by JelleZijlstra — APPROVED** [review]

File: stdlib/enum.pyi [hunk 2]
```diff
@@ -237,6 +237,8 @@ if sys.version_info >= (3, 11):
         _value_: str
         @_magic_enum_attr
         def value(self) -> str: ...
+        @staticmethod
+        def _generate_next_value_(name: str, start: Unused, count: Unused, last_values: Unused) -> str: ...
```
Comment (AlexWaygood) [inline stdlib/enum.pyi:241]: The idea of `_generate_next_value_` is that you can override it in subclasses of `Enum` to provide custom behaviour for how `auto()` works. As such, I don't think we should use `object` annotations (which is what `Unused` is) for the parameters, or subclassers won't get the correct signature if they do actually want to use the parameters.

```suggestion
        def _generate_next_value_(name: str, start: int, count: int, last_values: list[str]) -> str: ...
```

File: tests/stubtest_allowlists/py311.txt [hunk 3]
```diff
@@ -17,6 +17,8 @@ configparser.LegacyInterpolation.__init__
 configparser.ParsingError.filename
 enum.Enum.__init__
 enum.Enum._generate_next_value_
+# Python3.11 used to dynamically add this method:
```
Comment (AlexWaygood) [inline tests/stubtest_allowlists/py311.txt:20]:
```suggestion
# Not strictly speaking a staticmethod on 3.11, but it acts like one:
```

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**